src/actions/ProductsActions.ts

```diff
@@ -57,3 +69,23 @@ export const getProductsByProvider = (id: string) => {
       });
   });
 };
+
+export const getProductById = (id: string) => {
+  return new Promise<product>((resolve, reject) => {
+    useQuery<getProductByIdReturnTypes>(getProductByIdQuery, { id }).then(
+      ({ products }) => {
+        if (!products || !products[0]) {
+          Notify.create({
+            message: 'No se encontro el producto',
+            type: 'negative',
+          });
+
+          reject(null);
+          return null;
```

Comment:
Is this return necessary?

Reply:
I've double-checked and it's necessary. After the rejection, the code continues